### PR TITLE
refactor: solana sign and send transaction

### PR DESCRIPTION
**Pull request by zoruka**
# Description

This PR changes the logic for Solana WalletConnect connector to send the transaction for the wallet to sign and send instead of sending in the dapp. It goes alongside with https://github.com/WalletConnect/web-examples/pull/664.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-xxx
For GH issues: closes #...

# Showcase (Optional)

https://github.com/user-attachments/assets/bbfa4b01-aca4-4230-b5bf-f8f962cb0068

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [X] My changes generate no new warnings
- [X] I have reviewed my own code
- [X] I have filled out all required sections

### CODE REVIEW

**Comment by zoruka** [comment]
> eGetting the following error while calling `SIGN AND SEND TRANSACTION (WALLET)` example
> 
> ```js
> s.signAndSendTransaction is not a function
> ```
> 
> Steps to repro:
> 
> * Connect with Phantom extension
> * Make sure Phantom in test mode and you're in Solana Testnet in the extension
> * Switch to Solana Testnet
> * Try `SIGN AND SEND TRANSACTION (WALLET)` example
> 
> The `SIGN AND SEND TRANSACTION (DAPP)` example works.

Good catch @enesozturk, thanks! I have added it with [c741a8e](https://github.com/WalletConnect/web3modal/pull/2646/commits/c741a8e7b54f4a563626c8d1fb883e800f71c54a)

https://github.com/user-attachments/assets/1e12a513-4ebe-4b85-be97-e85a2414d93b

File: apps/laboratory/src/components/Solana/SolanaTests.tsx [hunk 1]
```diff
@@ -48,10 +49,16 @@ export function SolanaTests() {
...
+              Sign and Send Transaction (dApp)
             </Heading>
             <SolanaSendTransactionTest />
           </Box>
+          <Box>
+            <Heading size="xs" textTransform="uppercase" pb="2">
+              Sign and Send Transaction (Wallet)
+            </Heading>
+            <SolanaSignAndSendTransaction />
+          </Box>
```
Comment (tomiir) [inline apps/laboratory/src/components/Solana/SolanaTests.tsx:61]: What is the difference between Dapp and Wallet?

Comment (zoruka) [reply apps/laboratory/src/components/Solana/SolanaTests.tsx:61]: Dapp: the dapp request to wallet sign transaction and sends to blockchain
Wallet: the dapp request to wallet sign and send the transaction

Comment (tomiir) [reply apps/laboratory/src/components/Solana/SolanaTests.tsx:61]: Hmm I feel this is confusing, so you are doing basically

> Request wallet to sign transaction => Manually send it from lab
> Request wallet to sign and send transaction directly

Why send the tx from lab in the first place? isn't just requesting the signing enough? 

Just a nit tho

**Comment by tomiir** [comment]
Could sign a versioned tx with Phantom, but not a regular tx

**Review by enesozturk — APPROVED** [review]

**Review by tomiir — APPROVED** [review]

File: apps/laboratory/src/components/Solana/SolanaTests.tsx (hunk 1 above)
Comment (zoruka) [reply apps/laboratory/src/components/Solana/SolanaTests.tsx:61]: There is the two ways of doing it and both are valid: the first case is testing the `signTransaction` and `sendTransaction` functions, the second is testing `signAndSendTransaction` which sends for wallet handle the blockchain connection.

**Pull request merged**